nodestream/schema/migrations/auto_change_detector.py

```diff
@@ -0,0 +1,580 @@
+            renamed_types_location,
+            ignore_created_and_deleted=False,
+        ):
+            detect_changes_between_types(from_type_name, to_type_name)
+
+    def detect_property_changes(
+        self,
+        from_population: Dict[str, GraphObjectSchema],
+        to_population: Dict[str, GraphObjectSchema],
+        deleted_property_location: Set[Tuple[str, str]],
+        renamed_property_location: Set[Tuple[str, str, str]],
+        added_property_loction: Set[Tuple[str, str]],
+        deleted_types_location: Set[str],
+        renamed_types_location: Set[str],
+        new_types_location: Set[Tuple[str, str]],
+        properties_attr: str = "non_key_properties",
+    ):
+        def detect_changes_between_types(from_type_name, to_type_name):
+            from_type_props = getattr(from_population[from_type_name], properties_attr)
+            to_type_props = getattr(to_population[to_type_name], properties_attr)
+            # To start, we assume that all properties missing from one side or
+            # the other were created or deleted.
+            deleted = from_type_props - to_type_props
+            added = to_type_props - from_type_props
+            renamed = set()
+
+            # We don't have any smart way to ascertain what is probably
+            # a rename, so we are just going to have to ask about every
+            # combination of properties
+            for added_property in added:
+                for deleted_property in deleted:
+                    if self.input.ask_property_renamed(
+                        to_type_name, deleted_property, added_property
+                    ):
+                        renamed.add((to_type_name, deleted_property, added_property))
+
+            # For renamed properties, we can remove them from the
+            # added and removed sets and add it to to the final renamed set.
+            for item in renamed:
+                _, deleted_property, added_property = item
+                deleted.remove(deleted_property)
+                added.remove(added_property)
+                renamed_property_location.add(item)
+
+            # We can add the type and the property to the final set once
+            # we have removed renamed properties.
+            for property in added:
+                added_property_loction.add((to_type_name, property))
+            for property in deleted:
+                deleted_property_location.add((to_type_name, property))
+
+        for from_type_name, to_type_name in self.compare_types_pairwise(
+            from_population,
+            to_population,
+            deleted_types_location,
+            new_types_location,
+            renamed_types_location,
+        ):
+            detect_changes_between_types(from_type_name, to_type_name)
+
+    def detect_type_changes(
+        self,
+        from_population: Dict[str, GraphObjectSchema],
+        to_population: Dict[str, GraphObjectSchema],
+        deleted_types_location: Set[str],
```

Comment:
`_location` is confusing. This is the deleted_types set. Not sure what `_location` is signifying here other than by reference passed param?

Reply:
Hmm.. good point. Are you suggesting we call this simply `deleted_types`?  If so, seems reasonable to me.